docs/builtins.rst

```diff
@@ -0,0 +1,296 @@
+
+
+How to use Functions
+====================
+
+FTL Functions can only be called inside of placeables. Use them to return a
+value to be interpolated in the message or as selectors in select expressions.
+
+Example::
+
+
+  today-is = Today is { DATETIME($date) }
+
+
+Function parameters
+===================
+
+Functions may (but don't have to) accept positional and keyword arguments.
+Some parameters are only available to developers passing the Function as
+an argument.
+See the reference below for more information about the arguments accepted for
+each built-in function.
+
+
+Built-in Functions
+==================
+
+Built-in functions are very generic and should be applicable to any translation
+environment.
+
+``NUMBER``
+    Formats a number to a string in a given locale.
+
+    Example::
+
+      dpi-ratio = Your DPI ratio is { NUMBER($ratio, minimumFractionDigits=2) } 
+
+    Parameters::
+
+      ''currencyDisplay''
+      ''useGrouping''
+      ''minimumIntegerDigits''
+      ''minimumFractionDigits''
+      ''maximumFractionDigits''
+      ''minimumSignificantDigits''
+      ''maximumSignificantDigits''
+
+    Devloper parameters::
+
+      ''style''
+      ''currency''
+
+    See the `Intl.NumberFormat`_ for the description of the parameters.
+
+
+``DATETIME``
+    Formats a date to a string in a given locale.
+
+    Example::
+
+      today-is = Today is { DATETIME($date, month=long, year=numeric, day=numeric) } 
+
+    Parameters::
+
+      ''hour12''
+      ''weekday''
+      ''era''
+      ''year''
+      ''month''
+      ''day''
+      ''hour''
+      ''minute''
+      ''second''
+      ''timeZoneName''
+
+    Devloper parameters::
+
+      ''timeZone''
+
+    See the `Intl.DateTimeFormat`_ for the description of the parameters.
+
+``PLURAL``
+    Returns a plural form that matches the number for a given locale.
+
+    Example::
+  
+      liked-count = { PLURAL($num) ->
+        [one]   One person liked your message
+       *[other] { $num } people liked your message
+      }
+      
+    Parameters::
+
+      ''minimumIntegerDigits''
+      ''minimumFractionDigits''
+      ''maximumFractionDigits''
+      ''minimumSignificantDigits''
+      ''maximumSignificantDigits''
+
+    Devloper parameters::
+
+      ''type''
+
+    See the `Intl.PluralRules`_ for the description of the parameters.
+
+``LIST``
+    Formats a list to a string in a given locale.
+    Accepts the same parameters as `Intl.ListFormat`_ API.
+
+    Example::
+
+      users = { LIST($user1, $user2, $user3) }
+
+      users2 = { LIST($users) }
+
+    Parameters::
+
+      ''style''
+
+    Devloper parameters::
+
+      ''type''
+
+    See the `Intl.ListFormat`_ for the description of the parameters.
+
+``LEN``
+    Returns the number that represents the length of the list argument.
+    Similar to JS `Array.length`_ API.
+
+    Example::
+
+      unread-emails = Number of unread emails: { LEN($emails) }
+
+``TAKE``
+    Returns a slice of a list.
```

Comment:
`... starting from the first element`.  Please describe the parameters that `TAKE` and `DROP` accept.